pkg/appregistry/client.go

```diff
@@ -0,0 +1,105 @@
+package appregistry
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+// OperatorMetadata encapsulates operator metadata and manifest assocated with a package
+type OperatorMetadata struct {
+	// Namespace is the namespace in app registry server under which the package is hosted.
+	Namespace string
+
+	// Repository is the repository name for the specified package in app registry
+	Repository string
+
+	// Release represents the release or version number of the given package
+	Release string
+
+	// Digest is the sha256 hash value that uniquely corresponds to the blob associated with the release
+	Digest string
+
+	// Manifest encapsulates operator manifest
+	Manifest *Manifest
+}
+
+func (om *OperatorMetadata) ID() string {
+	return fmt.Sprintf("%s/%s", om.Namespace, om.Repository)
+}
+
+type client struct {
+	adapter      apprApiAdapter
+	decoder      blobDecoder
+	unmarshaller blobUnmarshaller
+}
+
+func (c *client) RetrieveAll() ([]*OperatorMetadata, error) {
+	packages, err := c.adapter.ListPackages()
+	if err != nil {
+		return nil, err
+	}
+
+	list := make([]*OperatorMetadata, len(packages))
```

Comment:
operatorMetaDataList probably would have made more sense.

Reply:
I have to implement a `operatorMetaDataList`. This is not a k8s type.